pkg/api/meta/builds.go

```diff
@@ -24,8 +24,8 @@ func NewBuildMutator(build *buildapi.Build) ImageReferenceMutator {
 
 func (m *buildSpecMutator) Mutate(fn ImageReferenceMutateFunc) field.ErrorList {
 	var errs field.ErrorList
-	for i, image := range m.spec.Source.Images {
-		if err := fn(&image.From); err != nil {
+	for i := range m.spec.Source.Images {
+		if err := fn(&m.spec.Source.Images[i].From); err != nil {
```

Comment:
@smarterclayton @csrwng fyi, this was never doing what it was supposed to do before, it was only mutating the copy of image that the range iterator was making.  yay go.